service/api_impl.ml

```diff
@@ -17,9 +18,33 @@ let make_commit ~engine ~owner ~name hash =
          let response, results = Service.Response.create Results.init_pointer in
          let arr = Results.jobs_init results (List.length jobs) in
          jobs
-         |> List.iteri (fun i (variant, outcome) ->
+         |> List.iteri (fun i (variant, outcome, ts) ->
                 let slot = Capnp.Array.get arr i in
                 Raw.Builder.JobInfo.variant_set slot variant;
+                let queued_at, started_at, finished_at =
+                  match ts with
+                  | None -> (None, None, None)
+                  | Some (Run_time.Queued v) -> (Some v, None, None)
+                  | Some (Running v) ->
+                      (Some v.queued_at, Some v.started_at, None)
+                  | Some (Finished v) ->
+                      (Some v.queued_at, v.started_at, Some v.finished_at)
+                in
+                let queued_at_t = Raw.Builder.JobInfo.queued_at_init slot in
+                let module S = Raw.Builder.JobInfo.QueuedAt in
+                (match queued_at with
+                | None -> S.none_set queued_at_t
+                | Some v -> S.ts_set queued_at_t v);
+                let started_at_t = Raw.Builder.JobInfo.started_at_init slot in
+                let module S = Raw.Builder.JobInfo.StartedAt in
+                (match started_at with
+                | None -> S.none_set started_at_t
+                | Some v -> S.ts_set started_at_t v);
+                let finished_at_t = Raw.Builder.JobInfo.finished_at_init slot in
+                let module S = Raw.Builder.JobInfo.FinishedAt in
+                (match finished_at with
+                | None -> S.none_set finished_at_t
+                | Some v -> S.ts_set finished_at_t v);
```

Comment:
Maybe it is worth being split in a `let f variant outcome t = (* ... *)` function above as it is a big function?